token-swap/program/src/processor.rs

```diff
@@ -103,11 +107,26 @@ impl State {
             &[],
             amount,
         )?;
-        invoke_signed(&ix, accounts, signers)
+        #[cfg(target_arch = "bpf")]
+        {
+            invoke_signed(&ix, accounts, signers)
+        }
+        #[cfg(not(target_arch = "bpf"))]
```

Comment:
Be nice to keep these featurizations out of the main code, can these be in an invoke_signed stub instead?